src/core/cgroup.c

```diff
@@ -3135,19 +3124,20 @@ int unit_get_memory_current(Unit *u, uint64_t *ret) {
         if (r < 0)
                 return r;
         if (r > 0)
-                r = cg_get_attribute("memory", u->cgroup_path, "memory.current", &v);
+                r = cg_get_attribute_as_uint64("memory", u->cgroup_path, "memory.current", &v);
         else
-                r = cg_get_attribute("memory", u->cgroup_path, "memory.usage_in_bytes", &v);
+                r = cg_get_attribute_as_uint64("memory", u->cgroup_path, "memory.usage_in_bytes", &v);
         if (r == -ENOENT)
                 return -ENODATA;
```

Comment:
... but the other ones seem to do that. Shouldn't this translation be folded into cg_get_attribute_as_uint64() to make things consistent and simplify all callers?

Reply:
Sure could do that.